src/DynamoCore/Scheduler/PreviewGraphAsyncTask.cs

```diff
@@ -55,13 +55,13 @@ internal List<Guid> Initialize(EngineController controller, WorkspaceModel works
             try
             {
                 engineController = controller;
-                TargetedWorkspace = workspace;                
-                modifiedNodes = ComputeModifiedNodes(workspace);                
-                previewGraphData = engineController.PreviewGraphSyncData(modifiedNodes,verboseLogging);
+                TargetedWorkspace = workspace;
+                modifiedNodes = ComputeModifiedNodes(workspace);
+                previewGraphData = engineController.PreviewGraphSyncData(modifiedNodes, verboseLogging);
```

Comment:
@aparajit-pratap @mjkkirschner @sm6srw @pinzart30 I have a question, if a graph(in manual mode) has been run once, then the user add a new node, what should this function return? I would expect this function to return me the new node's node id, but it is currently returning an empty set and causing a bug.

Reply:
was `PreviewGraphSyncData` returning empty list before your change to computeModifiedNodes?

Reply:
@mjkkirschner Previously the function is also returning empty. `modifiedNodes` were always all of the nodes in workspace so I would not trust the return value at all. As you can see from the comment from Ben, that TODO from 7 years ago never finished. Right now I am setting the modified nodes as an example, and in my case, I am adding a new node to the existing workspace, but this function does not return the expected results.

Reply:
Since the description of the `PreviewGraphSyncData` function is `This is called on the main thread from PreviewGraphSyncData to generate the list of node id's that will be executed on the next run`. I would expect it to return the id of newly added node.

Reply:
Thanks to Turbo Team who fixed this API, this is now working